sympy/series/series.py

```diff
@@ -10,3 +13,151 @@ def series(expr, x=None, x0=0, n=6, dir="+"):
+    -(x**2*exp(x) - E)**3*(-14*exp(3) - 14*E + 7 + 21*exp(2) + 7*exp(4))*exp(-4)/54 - 7*(x**2*exp(x) - E)**2*exp(-2)/54
+    + (x**2*exp(x) - E)*exp(-1)/3 + 1
+
+    Here the number of terms required is, by default, 3.
+
+    >>> lagrange_inversion_theorem(eq, x, 1)
+    -7*(x**2*exp(x) - E)**2*exp(-2)/54 + (x**2*exp(x) - E)*exp(-1)/3 + 1
+
+    Here the point with respect to which the expansion is found is changed from x=1 to x=2.
+
+    >>> lagrange_inversion_theorem(eq, x, 2)
+    -7*(x**2*exp(x) - 4*exp(2))**2*exp(-4)/512 + (x**2*exp(x) - 4*exp(2))*exp(-2)/8 + 2
+
+    >>> eq = exp(x)
+    >>> lagrange_inversion_theorem(eq, x, 2)
+    -(exp(x) - exp(2))**2*exp(-4)/2 + (exp(x) - exp(2))*exp(-2) + 2
+
+    Here the number of terms and the point with respect to which the inverse expansion will be calculated are both
+    not provided explicitly.
+
+    >>> lagrange_inversion_theorem(eq, x)
+    -(exp(x) - 1)**2/2 + exp(x) - 1
+
+    Note
+    ====
+
+    There is a straightforward derivation using complex analysis and contour integration;
+    the complex formal power series version is a consequence of knowing the formula for polynomials,
+    so the theory of analytic functions may be applied. Actually, the machinery from analytic
+    function theory enters only in a formal way in this proof, in that what is really needed is
+    some property of the formal residue, and a more direct formal proof is available.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Lagrange_inversion_theorem
+    .. [2] http://mathworld.wolfram.com/LagrangeInversionTheorem.html
+    .. [3] http://www.cfm.brown.edu/people/dobrush/am33/Mathematica/lit.html
+    """
+
+    try:
+        not eq.free_symbols
+    except AttributeError:
+        raise ValueError("The function provided is not analytic i.e. f' = 0. Lagrange's inversion theorem requires "
+                         "that the function be analytic")
+
+    if len(eq.free_symbols) == 1:
+        if x not in eq.free_symbols:
+            raise ValueError("The function variable provided does not match the equation variable")
+
+    w = x
+    eq = eq.subs(x, w)
+    z = eq
+
+    z_diff = diff(eq, w)
+
+    if z_diff.subs(w, a) == 0 or not (z_diff.subs(w, a)).is_real:
+        raise ValueError("The first derivative of the function is zero at the point 'a' provided i.e. f'(a) = 0. "
+                         "Lagrange's inversion theorem requires that the derivative of the function be non zero at "
+                         "the point 'a'.")
+
+    f_a = z.subs(w, a)
+    result = a
+
+    if n == 1:
+        return result
+
+    a_1 = ((w - a)/(z - f_a))
+    l = limit(a_1, w, a)
+    result = result + l*(z - f_a)
+
+    if n == 2:
+        return result
+
+    a_2 = diff(((w - a)/(z - f_a))**2, w, 1)
+
+    for i in range(2, n):
+        l = limit(a_2, w, a)
+        result = result + (((z - f_a) ** i) / factorial(i)) * l
+        a_2 = (i+1) * ((w - a / z - f_a) ** i) * a_2
```

Comment:
@sylee957  First of all, I'm sorry for being so late to reply. I had my exams for the past 2 weeks and was prepping for them. 
Turns out, I didn't implement the derivative correctly (and so the future result which uses the derivative also turned out to be incorrect) as you pointed out. It was correct initially. While trying to make the previously calculated derivative useful, which would have speed ed up the entire process drastically, I miscalculated the way the derivative would be used (my bad here). Turns out that using the previously calculated derivative, is not as easy as it sounds because it is used in products which increase in number drastically as `n` increases. So taking the previously calculated derivative, is not an option.
Fortunately the correct results are displayed now but of course it takes up time.
So far I haven't been able to figure out a way to re-use the derivative. If you can help here it'll be more than great. 
(I've considered using the option of a dictionary of sorts but in that approach, since the derivative is reused a lot of times - in big `n`s, re-iterating the dictionary every single time, will be a costly operation. Also, we'll have to find a way to test at every point of the calculation if it's derivative would be available in the dictionary or not. This again, would be costly.)